File: Data Structures & Algorithms/majority-element/submission-0.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:

        # since majority element has to occur more than floor(n/2)
        # times

        # other O(n) approach would be to store and iterate through
        # hashmap

        cnt = 0
        res = 0
        for num in nums:
            if cnt == 0:
                res = num
            if num == res:
                cnt += 1
            else:
                cnt -= 1
        return res

File: Data Structures & Algorithms/is-anagram/submission-0.cpp
#include <array> 

class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) return false;
        std::array<int, 26> cnt = {0};
        for (const auto& c : s) cnt[c - 'a']+=1;
        for (const auto& c: t){
            if (--cnt[c-'a'] == -1) return false;
        }
        return true;
    }
};

